src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -425,13 +426,46 @@ protected function executeStatement(string $sql, array $parameters = [], array $
             // create table
             if ($this->autoSetup) {
                 $this->setup();
+
+                $stmt = $this->driverConnection->executeStatement($sql, $parameters, $types);
+            } else {
+                throw $e;
             }
-            $stmt = $this->driverConnection->executeStatement($sql, $parameters, $types);
         }
 
         return $stmt;
     }
 
+    private function executeInsert(string $sql, array $parameters = [], array $types = []): string
+    {
+        $this->driverConnection->beginTransaction();
+
+        try {
+            try {
+                $this->driverConnection->executeStatement($sql, $parameters, $types);
+            } catch (TableNotFoundException $e) {
+                // create table
+                if ($this->autoSetup) {
+                    $this->setup();
+
+                    $this->driverConnection->executeStatement($sql, $parameters, $types);
+                } else {
+                    throw $e;
+                }
+            }
+
+            $id = $this->driverConnection->lastInsertId();
+
+            if (false === $id) {
+                throw new TransportException('lastInsertId() returned false, no id was returned.');
+            }
+        } finally {
+            $this->driverConnection->commit();
+        }
+
+        return (string) $id;
```

Comment:
```suggestion
        return $id;
```

Reply:
When I remove this casting, psalm complains. Suggestions?

Reply:
ignore psalm: it's just a tool that should trigger our brain, and here we know better than the tool

Reply:
:+1: fixed!